drivers/flash/flash_stm32f4x.c

```diff
@@ -20,6 +20,17 @@ bool flash_stm32_valid_range(struct device *dev, off_t offset, u32_t len,
 {
 	ARG_UNUSED(write);
 
+#if (FLASH_SECTOR_TOTAL == 12) && defined(FLASH_OPTCR_DB1M)
+	struct stm32f4x_flash *regs = FLASH_STM32_REGS(dev);
+	/*
+	 * RM0090, table 7.1: STM32F42xxx, STM32F43xxx
+	 */
+	if (regs->optcr & FLASH_OPTCR_DB1M) {
+		/* Device configured in Dual Bank, but not supported for now */
+		return -ENOTSUP;
```

Comment:
The function prototype says you should return a bool, not an int

Reply:
Changing this to -1 is still returning an int

Reply:
I should not hack during meetings... Thanks